### PR TITLE
backend-common: Remove read method from UrlReader

### PR DESCRIPTION
**Pull request by jhaals**
**BREAKING**: Removed deprecated `read` method from the `UrlReader` interface. All implementations should use the `readUrl` method instead.

Migrated `UrlReader` and related types to `backend/backend-plugin-api`, types remain re-exported from `backend-common` for now.

### CODE REVIEW

File: packages/backend-plugin-api/src/services/definitions/urlReaderServiceRef.ts [hunk 1]
```diff
@@ -15,10 +15,278 @@
...
+export type UrlReaderService = {
+  /**
+   * Reads a single file and return its content.
+   *
+   * @remarks
+   *
+   * This is a replacement for the read method that supports options and
+   * complex responses.
+   *
+   * Use this as the read method will be removed in a future release.
```
Comment (OrkoHunter) [inline packages/backend-plugin-api/src/services/definitions/urlReaderServiceRef.ts:34]: It's gone now! 😁

**Review by OrkoHunter — COMMENTED** [review]
Nice! This [docs page](https://backstage.io/docs/plugins/url-reader) will need some of line removals too. :)

File: packages/backend-plugin-api/src/services/definitions/urlReaderServiceRef.ts (hunk 1 above)
Comment (jhaals) [reply packages/backend-plugin-api/src/services/definitions/urlReaderServiceRef.ts:34]: Thanks! ❤️

**Comment by jhaals** [comment]
Thanks @OrkoHunter for pointing out the docs and tsdoc, I've addressed your comments

**Review by iamEAP — APPROVED** [review]

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nice! 😁 

Just need a changeset for `@backstage/plugin-catalog-backend-module-openapi`

**Review by Rugvip — APPROVED** [review]
:shipit:

**Pull request merged**